lib/stsci/tools/stpyfits.py

```diff
@@ -75,15 +78,27 @@ class _ConstantValueImageBaseHDU(fits.hdu.image._ImageBaseHDU):
     """
 
     __doc__ += fits.hdu.image._ImageBaseHDU.__doc__
+    _header = Header()
 
     def __init__(self, data=None, header=None, do_not_scale_image_data=False,
                  uint=False, **kwargs):
         if header and 'PIXVALUE' in header and header['NAXIS'] == 0:
-            header = header.copy()
+            if not ASTROPY_VER_GE32:
```

Comment:
I haven't had a careful look at this, but an alterative to putting this `if` statement might be, at least until @saimn's `_BaseHeader` grows a `.copy()` method, would be to wrap the header in an adapter class which provides it.  That might be more transparent.  I'm sorry if this is a vague comment though. Like I said, not fully thought through.  That's just usually what I end up doing in cases like this.